src/hipscat/io/validation.py

```diff
@@ -16,13 +33,129 @@ def is_valid_catalog(pointer: FilePointer, storage_options: Union[Dict[Any, Any]
+
+    ## Compare the pixels in _metadata with partition_info.csv
+    metadata_file = get_parquet_metadata_pointer(pointer)
+
+    # Use both strategies of reading the partition info: strict and !strict.
+    metadata_pixels = sort_pixels(
+        PartitionInfo.read_from_file(
+            metadata_file, strict=True, storage_options=storage_options
+        ).get_healpix_pixels()
+    )
+    if not np.array_equal(expected_pixels, metadata_pixels):
+        handle_error("Partition pixels differ between catalog and _metadata file (strict)")
+
+    metadata_pixels = sort_pixels(
+        PartitionInfo.read_from_file(
+            metadata_file, strict=False, storage_options=storage_options
+        ).get_healpix_pixels()
+    )
+    if not np.array_equal(expected_pixels, metadata_pixels):
+        handle_error("Partition pixels differ between catalog and _metadata file (non-strict)")
+
+    partition_info_file = get_partition_info_pointer(pointer)
+    csv_pixels = sort_pixels(PartitionInfo.read_from_csv(partition_info_file).get_healpix_pixels())
+    if not np.array_equal(expected_pixels, csv_pixels):
+        handle_error("Partition pixels differ between catalog and partition_info.csv file")
+
+    ## Load as parquet dataset. Allow errors, and check pixel set against _metadata
+    ignore_prefixes = [
+        "_common_metadata",
+        "_metadata",
+        "catalog_info.json",
+        "provenance_info.json",
+        "partition_info.csv",
+        "point_map.fits",
+        "README",
+    ]
+
+    # As a side effect, this confirms that we can load the directory as a valid dataset.
+    (dataset_path, dataset) = read_parquet_dataset(
+        pointer,
+        ignore_prefixes=ignore_prefixes,
+        exclude_invalid_files=False,
+        storage_options=storage_options,
     )
 
+    parquet_path_pixels = []
+    for hips_file in dataset.files:
+        relative_path = hips_file[len(dataset_path) :]
+        healpix_pixel = get_healpix_from_path(relative_path)
+        if healpix_pixel == INVALID_PIXEL:
+            handle_error(f"Could not derive partition pixel from parquet path: {relative_path}")
 
-def is_catalog_info_valid(pointer: FilePointer, storage_options: Union[Dict[Any, Any], None] = None) -> bool:
+        parquet_path_pixels.append(healpix_pixel)
+
+    parquet_path_pixels = sort_pixels(parquet_path_pixels)
+
+    if not np.array_equal(expected_pixels, parquet_path_pixels):
+        handle_error("Partition pixels differ between catalog and parquet paths")
+
+    if verbose:
+        # Print a few more stats
+        pixel_orders = [p.order for p in expected_pixels]
+        cov_order, cov_count = np.unique(pixel_orders, return_counts=True)
+        area_by_order = [hp.nside2pixarea(hp.order2nside(order), degrees=True) for order in cov_order]
+        total_area = (area_by_order * cov_count).sum()
+        print(
+            f"Approximate coverage is {total_area:0.2f} sq deg, or {total_area/41253*100:0.2f} % of the sky."
```

Comment:
Can we store the total area of the sky in a constant variable somewhere for future use? 😄

Reply:
I'm not sure about the use case for doing something like that (and could always be computed from the list of partitions (or from the MOC)). If you've got more ideas, feel free to open up a new issue ;)